DSMRloggerAPI.ino

```diff
@@ -380,117 +384,145 @@ void setup()
   sprintf(cMsg, "Last reset reason: [%s]\r", ESP.getResetReason().c_str());
   DebugTln(cMsg);
 
-  telegramInterval = millis() + 5000;
-  noMeterWait      = millis() + 5000;
-  upTimeSeconds    = (millis() / 1000) + 50;
-  nextSecond       = millis() + 1000;
-
 #if defined( HAS_OLED_SSD1306 ) || defined( HAS_OLED_SH1106 )
     oled_Print_Msg(0, "<DSMRloggerAPI>", 0);
     oled_Print_Msg(1, "Startup complete", 0);
     oled_Print_Msg(2, "Wait for first", 0);
     oled_Print_Msg(3, "telegram .....", 500);
 #endif  // has_oled_ssd1306
-  
-} // setup()
 
 
+} // setup()
+
 //===========================================================================================
-void loop () 
+//===[ blink status led in ms ]===
+void blinkLEDms(uint32_t iDelay){
+  //blink the statusled, when time passed... #non-blocking blink
+  DECLARE_TIMER_MS(timerBlink, iDelay);
+  CHANGE_INTERVAL_MS(timerBlink, iDelay);
+  if (DUE(timerBlink))
+    blinkLEDnow();
+}
+
+//===[ blink status now ]===
+void blinkLEDnow(){
+    pinMode(LED_BUILTIN, OUTPUT);
+    digitalWrite(LED_BUILTIN, !digitalRead(LED_BUILTIN));
+}
+
+//===[ no-blocking delay with running background tasks in ms ]===
+void delayms(unsigned long delay_ms)
 {
-  httpServer.handleClient();
-  MDNS.update();
-  handleKeyInput();
-  //handleRefresh();  // webSocket
-  handleMQTT();
-  handleMindergas();
+  DECLARE_TIMER_MS(timer, delay_ms);
+  CHANGE_INTERVAL_MS(timer, delay_ms);
+  while (!DUE(timer))
+    doBackgroundTasks();
```

Comment:
Hier gaat bij mij het licht helemaal uit!
Waarom is doBackgroundTask() afhankelijk van “timer”?

Wellicht dat ik de context mis omdat ik niet zie wat er vóór regel 418 en ná regel 419 staat ..